File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/java_to_py/Order.py
class Order:
    def __init__(self):
        self.menu = []
        self.selected_dishes = []
        self.sales = {}

    def add_dish(self, dish):
        if dish is None or 'dish' not in dish or 'price' not in dish or 'count' not in dish:
            return False
        
        for menu_dish in self.menu:
            if dish['dish'] == menu_dish['dish']:
                dish_count = dish['count']
                menu_dish_count = menu_dish['count']
                
                if menu_dish_count < dish_count:
                    return False
                
                menu_dish['count'] -= dish_count
                self.selected_dishes.append(dish)
                return True
        
        return False

    def calculate_total(self):
        total = 0.0
        for dish in self.selected_dishes:
            dish_name = dish['dish']
            price = dish['price']
            count = dish['count']
            sale = self.sales.get(dish_name, 1.0)
            total += price * count * sale
        return total

    def checkout(self):
        if not self.selected_dishes:
            return False
        total = self.calculate_total()
        self.selected_dishes.clear()
        return total

File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_cpp/code_output/Order.cpp
#include <vector>
#include <map>
#include <unordered_map>
#include <string>
#include <any>

class Order {
public:
    std::vector<std::map<std::string, std::any>> menu;
    std::vector<std::map<std::string, std::any>> selectedDishes;
    std::unordered_map<std::string, double> sales;

    Order() = default;

    bool addDish(const std::map<std::string, std::any>& dish) {
        if (dish.find("dish") == dish.end() ||
            dish.find("price") == dish.end() ||
            dish.find("count") == dish.end()) {
            return false;
        }

        for (auto& menuDish : menu) {
            if (std::any_cast<std::string>(dish.at("dish")) ==
                std::any_cast<std::string>(menuDish.at("dish"))) {
                int dishCount = std::any_cast<int>(dish.at("count"));
                int menuDishCount = std::any_cast<int>(menuDish.at("count"));

                if (menuDishCount < dishCount) {
                    return false;
                } else {
                    menuDish["count"] = menuDishCount - dishCount;
                    selectedDishes.push_back(dish);
                    return true;
                }
            }
        }
        return false;
    }

    double calculateTotal() {
        double total = 0;
        for (auto& dish : selectedDishes) {
            std::string dishName = std::any_cast<std::string>(dish.at("dish"));
            double price = std::any_cast<double>(dish.at("price"));
            int count = std::any_cast<int>(dish.at("count"));
            double sale = 1.0;
            auto it = sales.find(dishName);
            if (it != sales.end()) {
                sale = it->second;
            }
            total += price * count * sale;
        }
        return total;
    }

    std::any checkout() {
        if (selectedDishes.empty()) {
            return false;
        }
        double total = calculateTotal();
        selectedDishes.clear();
        return total;
    }
};